internal/transport/transport_test.go

```diff
@@ -536,6 +536,40 @@ func (s) TestInflightStreamClosing(t *testing.T) {
 	}
 }
 
+// TestClientTransportDrainsAfterStreamIdExhausted tests that when
```

Comment:
FWIW this style of comment, duplicating the function name, is unnecessary for tests.